libraries/Microsoft.Bot.Builder.AI.QnA/Utils/QnACardBuilder.cs

```diff
@@ -77,48 +77,82 @@ public static IMessageActivity GetSuggestionsCard(List<string> suggestionsList,
         /// </summary>
         /// <param name="result">Result to be dispalyed as prompts.</param>
-        /// <param name="cardNoMatchText">No match text.</param>
+        /// <param name="cardNoMatchText">No match text.</param>       
         /// <returns>IMessageActivity.</returns>
         public static IMessageActivity GetQnAPromptsCard(QueryResult result, string cardNoMatchText)
+        {
+            return GetQnADefaultResponse(result, true);
+        }
+
+        /// <summary>
+        /// Get Card for Default QnA Maker scenario.
+        /// </summary>
+        /// <param name="result">Result to be dispalyed as prompts.</param>
+        /// <param name="displayPreciseAnswerOnly">renderingchoice.</param>
+        /// <returns>IMessageActivity.</returns>
+        public static IMessageActivity GetQnADefaultResponse(QueryResult result, bool displayPreciseAnswerOnly)
         {
             if (result == null)
             {
                 throw new ArgumentNullException(nameof(result));
             }
 
-            if (cardNoMatchText == null)
-            {
-                throw new ArgumentNullException(nameof(cardNoMatchText));
-            }
-
             var chatActivity = Activity.CreateMessageActivity();
             chatActivity.Text = result.Answer;
-            var buttonList = new List<CardAction>();
+            
+            List<CardAction> buttonList = null;
+            if (result?.Context?.Prompts != null &&
+                result.Context.Prompts.Any())
+            {
+                buttonList = new List<CardAction>();
+
+                // Add all prompt
+                foreach (var prompt in result.Context.Prompts)
+                {
+                    buttonList.Add(
+                        new CardAction()
+                        {
+                            Value = prompt.DisplayText,
+                            Type = "imBack",
+                            Title = prompt.DisplayText,
+                        });
+                }
+            }
 
-            // Add all prompt
-            foreach (var prompt in result.Context.Prompts)
+            string heroCardText = null;
+            if (!string.IsNullOrWhiteSpace(result?.AnswerSpan?.Text))
             {
-                buttonList.Add(
-                    new CardAction()
-                    {
-                        Value = prompt.DisplayText,
-                        Type = "imBack",
-                        Title = prompt.DisplayText,
-                    });
+                chatActivity.Text = result.AnswerSpan.Text;
+
+                // For content choice Precise only
+                if (!displayPreciseAnswerOnly)
+                {
+                    heroCardText = result.Answer;
+                }
             }
 
-            var plCard = new HeroCard()
+            if (buttonList != null || !string.IsNullOrWhiteSpace(heroCardText))
```

Comment:
So. Many. Null. Checks.

Reply:
We had to check as the flow has multiple scenarios

Reply:
We had to check as the flow has multiple scenarios.

---
In reply to: [430561991](https://github.com/microsoft/botbuilder-dotnet/pull/3935#discussion_r430561991) [](ancestors = 430561991)